### ISSUE TITLE
imagesx & imagesy not functioning (bad parameters)

### ISSUE THREAD

**Issue by barrychapman** [jrbasso/MeioUpload#16]
You are trying to use imagesx & imagesy on a string filename. You need to do the following:

```
function __createImage ( $image ) {

    $imagepath = $image['tmp_name'];
    $ext       = end(explode('.', $image['name'] ) );

    switch( $ext ) {
        case "png":
            $newImage = imagecreatefrompng( $imagepath );
            break;
        case "bmp":
            $newImage = imagecreatefromwbmp( $imagepath );
            break;
        case "jpg":
            $newImage = imagecreatefromjpeg( $imagepath );
            break;
        case "gif":
            $newImage = imagecreatefromgif( $imagepath );
            break;
    }

    return $newImage;       
}
```

/////////

```
function uploadCheckMaxWidth(&$model, $data) {
    foreach ($data as $fieldName => $field) {
        if (!$this->__model->validate[$fieldName]['MaxWidth']['check']) {
            return true;
        }
        $options = $this->__fields[$fieldName];

        $image = $this->__createImage( $field );    // Note we are creating an image object

        if (!empty($field['name']) && $options['length']['maxWidth'] > 0 && imagesx( $image ) > $options['length']['maxWidth']) {
            return false;
        }
    }
    return true;
}
```

**Comment by jrbasso** [comment]
Fix [min|max][width|height] validations. Closed by f22247030c01921780fb8d75cf0665c2257ce74b.

**Comment by jrbasso** [comment]
@barry thanks for your code, but I implement in different approach.
